newrelic/hooks/mlmodel_langchain.py

```diff
@@ -664,24 +672,34 @@ def create_chat_completion_message_event(
 
 
 def wrap_on_chain_start(wrapped, instance, args, kwargs):
-    run_manager = wrapped(*args, **kwargs)
     transaction = current_transaction()
     if not transaction:
-        return run_manager
-    # Only capture the first run_id.
-    if not hasattr(transaction, "_nr_chain_run_id"):
-        transaction._nr_chain_run_id = getattr(run_manager, "run_id", "")
+        return wrapped(*args, **kwargs)
```

Comment:
Is there a need to implement this logic for sync runs? If the `completion_id` storing/ checks are done in the async run wrapper, can we avoid changing the sync logic here?

Reply:
Yeah as far as I know monkeypatch could be used to override the function but I wouldn't be able to get the side_effect behavior that I have here-it would be a lot more awkward to implement.

Reply:
I suppose we don't have to change the sync logic. It seemed better design wise to just do the same thing for both async and sync though so they maintain symmetry.

Reply:
The current implementation shares the same _get_run_manager_info function which we wouldn't be able to do if they had different behaviors.